TreeMaker/interface/TreeMaker.h

```diff
@@ -376,24 +376,24 @@ class TreeNestedVector : public TreeObject<std::vector<std::vector<Base>>> {
 
 		// Constructor
 		TreeNestedVector() : TreeObject<Top>() {}
-		TreeNestedVector(string tempFull_, string title_="", bool nestedVectors_=true, bool associated_=false, int splitLevel_=0) :
-			TreeObject<Top>(tempFull_,title_,splitLevel_), nestedVectors(nestedVectors_), associated(associated_) {}
+		TreeNestedVector(string tempFull_, string title_="", bool nestedVectors_=true, bool nestedCounts_=true, bool associated_=false, int splitLevel_=0) :
+			TreeObject<Top>(tempFull_,title_,splitLevel_), nestedVectors(nestedVectors_), nestedCounts(nestedCounts_), associated(associated_) {}
 		// Destructor
 		~TreeNestedVector() override {}
 		
 		// Functions
 		// From: https://stackoverflow.com/questions/17294629/merging-flattening-sub-vectors-into-a-single-vector-c-converting-2d-to-1d
 		void flatten(Top const& all, Sub &accum, vector<int> &offsets) {
 			// Don't store any offsets if there are no sub-vectors
-			if (!associated && all.size() > 0)	offsets.insert(std::end(offsets),0);
+			if (!associated && all.size() > 0){ if(!nestedCounts) offsets.insert(std::end(offsets),0); }
```

Comment:
simplify this to one if condition: `if (!associated && all.size() > 0 && !nestedCounts)`

Reply:
for `nestedCounts` the `0` shouldn't be inserted, but the `else return` should be skipped, hence this ugly double `if`. The alternative would be an `else if (nestedCounts) ; else ...`?

Reply:
ah, then it's fine as is.